dotcom-rendering/src/components/YoutubeAtom/YoutubeAtomPlayer.tsx

```diff
@@ -444,23 +448,34 @@ export const YoutubeAtomPlayer = ({
 						playerVars: {
 							controls: 1,
 							// @ts-expect-error -- advised by YouTube for Android but does not exist in @types/youtube
-							external_fullscreen: 1,
+							external_fullscreen:
+								renderingTarget === 'Apps' ? 1 : 0,
 							fs: 1,
 							modestbranding: 1,
 							origin,
 							playsinline: 1,
 							rel: 0,
 						},
-						// Since IMA ads the YouTube player API no longer accepts ad configuration
 						embedConfig: {
 							relatedChannels: [],
+							// Since IMA ads the YouTube player API no longer accepts ad configuration
 							adsConfig: { disableAds: true },
 							enableIma: enableAds,
 							// YouTube recommends disabling related videos when IMA ads are enabled
 							disableRelatedVideos: enableAds,
 						},
 						events: {
 							onStateChange: onStateChangeListener,
+							onFullscreenToggled: () => {
+								if (renderingTarget === 'Apps') {
+									log('dotcom', {
+										from: 'YoutubeAtomPlayer fullscreen',
+										videoId,
+									});
+									// For Android only, iOS will stub the method
+									void getVideoClient().fullscreen();
```

Comment:
Given this says "toggle", will this get called in both directions? So when fullscreen is exited as well as entered?

If so, will the Android app understand what to do if we call the same `fullscreen` method in both cases?

Reply:
Spoke to @ab-gnm about this. You're right that the listener is a toggle which is notified of a fullscreen control press i.e. enter or exit.

However we expect the listener in practice to only signal the 'enter fullscreen' event. This is because when fullscreen is entered on Android the native layer will enter fullscreen and a native control will be shown to exit fullscreen.

We also set `external_fullscreen` for apps clients (but only enabled for Android) which means YouTube not perform any fullscreen behaviour itself.

We are going to test this once the native layer integration is complete and adjust if we need to.